docs/reference/mapping.asciidoc

```diff
@@ -123,44 +123,159 @@ You can create field mappings when you
 fields to an existing index with the <<indices-put-mapping,PUT mapping API>>.
 
 [float]
-== Updating existing field mappings
+[[create-mapping]]
+== Create an index with an explicit mapping
 
-Other than where documented, *existing field mappings cannot be
-updated*. Changing the mapping would mean invalidating already indexed
-documents. Instead, you should create a new index with the correct mappings
-and <<docs-reindex,reindex>> your data into that index. If you only wish
-to rename a field and not change its mappings, it may make sense to introduce
-an <<alias, `alias`>> field.
+You can use the <<indices-create-index,create index>> API to create a new index
+with an explicit mapping.
+
+[source,js]
+----
+PUT my-index
+{
+  "mappings": {
+    "properties": {
+      "age":    { "type": "integer" },  <1>
+      "email":  { "type": "keyword"  }, <2>
+      "name":   { "type": "text"  }     <3>
+    }
+  }
+}
+----
+// CONSOLE
+
+<1> Creates `age`, an <<number,`integer`>> field
+<2> Creates `email`, a <<keyword,`keyword`>> field
+<3> Creates `name`, a <<text,`text`>> field
 
 [float]
-== Example mapping
+[[add-field-mapping]]
+== Add a field to an existing mapping
 
-A mapping can be specified when creating an index, as follows:
+You can use the <<indices-put-mapping, put mapping>> API to add one or more new
+fields to an existing index.
+
+The following example adds `employee-id`, a `keyword` field with an
+<<mapping-index,`index`>> mapping parameter value of `false`. This means values
+for the `employee-id` field are stored but not indexed or available for search.
 
 [source,js]
----------------------------------------
-PUT my_index <1>
+----
+PUT /my-index/_mapping
 {
-  "mappings": {
-    "properties": { <2>
-      "title":    { "type": "text"  }, <3>
-      "name":     { "type": "text"  }, <4>
-      "age":      { "type": "integer" },  <5>
-      "created":  {
-        "type":   "date", <6>
-        "format": "strict_date_optional_time||epoch_millis"
+  "properties": {
+    "employee-id": {
+      "type": "keyword",
+      "index": false
+    }
+  }
+}
+----
+// CONSOLE
+// TEST[continued]
+
+[float]
+[[update-mapping]]
+== Update the mapping of a field
```

Comment:
Should this actually be a subsection of "Add a field to an existing mapping"? As it's currently formatted, it's at the same level as other API examples, but it seems more like an important note about the limitations of 'put mapping'.

Reply:
Although they use the same mechanism (the put mapping API), my thought here was that users could see "adding a field" and "updating an existing field" as two separate tasks.

However, I could also see this as a subheading. I incremented the heading with cbe9827.